File: acousticslib_matlab/crossspectraldensity2timeseries.m
function [time_series, t_range] = crossspectraldensity2timeseries(Sxy, dt)
N = length(Sxy);
shift_amount = 1 - ceil(N/2);
Sxy = circshift(Sxy, shift_amount);

time_series = ifft(Sxy)/dt;
time_series = circshift(time_series, shift_amount);

t_range = (0:(N - 1)).*dt;
t_range = t_range - (ceil(N/2) - 1)*dt;
end

File: acousticslib/cross_spectral_density_2_time_series.py
import numpy as np
from math import ceil

def time_series(Sxy,dt):
    N = len(Sxy)
    shift_amount = 1-ceil(N/2)
    Sxy = np.roll(Sxy, shift_amount)

    time_series = np.fft.ifft(Sxy,dt)
    time_series = np.roll(time_series,shift_amount)

    t_range = np.arange(0,N)*dt
    t_range = t_range - (ceil(N/2)-1)*dt

    return(time_series,t_range)